corehq/apps/hqcase/tasks.py

```diff
@@ -22,11 +23,13 @@ def explode_cases(user_id, domain, factor, task=None):
     old_to_new = dict()
     child_cases = list()
 
+    case_ids = get_case_ids_in_domain_by_owner(
```

Comment:
This is a good point, and I've already thought about it a bit, so let me lay out my thinking, and see if you agree.

Something I've been seeing a lot of in my refactoring is functions that have inconsistent return types—this is really common because CommCareCase.view has lots of different return types (you can use it to return counts, or whole docs, or just ids and keys, etc.), and nearly every function wrapping a view just pushes that outwards. This means that these functions become pretty difficult to reason with. If you see `get_me_foo(bar, wrap=True)`, ok sure you know that returns wrapped documents—but what about `get_me_foo(bar, wrap=wrap)`? (And I assure you, someone's going to call it that way!)

The other thing to note is that the pattern of getting the list of ids and then using `iter_docs`, while slightly less efficient for small queries (first order factor being the number of HTTP requests to couch), is a lot more memory efficient for large numbers of cases. In general I think we should be encouraging idioms that favor performance on large domains (which of course starts with **not** iterating through all cases, but also if you have to, doing it this way).

I think what I've been doing, in some places, inconsistently, is making separate functions for separate return types:
- count: `get_number_of_*s_*`
- ids: `get_*_ids_`
- full json: `get_*_docs_*`
- wrapped json: `get_*s_*`

Like I said, I haven't been super consistent about this. Let me know if you have suggestions for a better convention!